src/Data/Machine/Mealy.purs

```diff
@@ -48,11 +49,24 @@ import Effect.Class (class MonadEffect, liftEffect)
 
 newtype MealyT f s a = MealyT (s -> f (Step f s a))
 
+-- |let's you transform a mealy machine running in the context of `f` into one running in `g`
+-- if you can give a natural transformation from `f` to `g`
+-- 
+-- this can be useful if you use monad-transformers for example
```

Comment:
In PS, comments need to start with `|` before every line, to be included in the generated docs.